internal/container/deviceservice.go

```diff
@@ -39,6 +42,15 @@ func ProtocolDiscoveryFrom(get di.Get) sdkModels.ProtocolDiscovery {
 	return nil
 }
 
+// ProtocolValidatorFrom helper function queries the DIC and returns ProtocolProperty validator implementation.
+func ProtocolValidatorFrom(get di.Get) sdkModels.ProtocolValidator {
```

Comment:
```suggestion
func DeviceValidatorFrom(get di.Get) sdkModels.DeviceValidator {
```

Also adjust name in comments